src/Dependency/Hydrator/DependencyHydrator.php

```diff
@@ -16,53 +16,25 @@
 
 namespace Pimcore\Bundle\StudioBackendBundle\Dependency\Hydrator;
 
-use Pimcore\Bundle\StaticResolverBundle\Models\Element\ServiceResolverInterface;
+use Pimcore\Bundle\GenericDataIndexBundle\Model\Search\Interfaces\ElementSearchResultItemInterface;
 use Pimcore\Bundle\StudioBackendBundle\Dependency\Schema\Dependency;
-use Pimcore\Bundle\StudioBackendBundle\Util\Traits\ElementProviderTrait;
 
 /**
  * @internal
  */
 final readonly class DependencyHydrator implements DependencyHydratorInterface
 {
-    use ElementProviderTrait;
-
-    public function __construct(
-        private ServiceResolverInterface $serviceResolver,
-    ) {
-
-    }
-
-    public function hydrate(array $dependency): ?Dependency
+    public function hydrate(ElementSearchResultItemInterface $dependency): ?Dependency
```

Comment:
```suggestion
    public function hydrate(ElementSearchResultItemInterface $dependency): Dependency
```

nullable can be removed?

Reply:
I removed it. Not sure why it was there in the first place. It is like the permissions. Why would we get an empty dependency from the core, or an empty permissions.